server/services/generateReportService.ts

```diff
@@ -14,99 +14,208 @@
  */
 
 //@ts-ignore
-import { Legacy, Logger, PluginInitializerContext } from 'kibana';
+import { Legacy, PluginInitializerContext } from 'kibana';
+import { Logger } from '../../../../src/core/server/logging';
 import { Server, Request, ResponseToolkit } from 'hapi';
 import { RequestParams } from '@elastic/elasticsearch';
 import puppeteer from 'puppeteer';
-import imagesToPdf from 'images-to-pdf';
-import fs from 'fs';
 import { v1 as uuidv1 } from 'uuid';
 import { ServerResponse } from '../models/types';
 import { CLUSTER, FORMAT } from '../utils/constants';
+import { SearchResponse, GetResponse } from 'elasticsearch';
+import { Readable } from 'stream';
 
 type ElasticsearchPlugin = Legacy.Plugins.elasticsearch.Plugin;
 
 export default class GenerateReportService {
   esDriver: ElasticsearchPlugin;
-  private readonly log: Logger;
+  logger: Logger;
 
   constructor(
     private readonly initializerContext: PluginInitializerContext,
     esDriver: ElasticsearchPlugin
   ) {
     this.esDriver = esDriver;
-    this.log = this.initializerContext.logger.get();
+    this.logger = this.initializerContext.logger.get();
   }
 
   report = async (req: Request, h: ResponseToolkit): Promise<any> => {
     try {
       const {
         url,
         itemName,
+        source,
         reportFormat,
         windowWidth = 1200,
         windowLength = 800,
       } = req.payload as {
         url: string;
         itemName: string;
+        source: string;
         reportFormat: string;
-        windowWidth: number;
-        windowLength: number;
+        windowWidth?: number;
+        windowLength?: number;
```

Comment:
Do we need any specific validation our widht and height ? Such as some ration to ensure we don't mess the report ?

Reply:
Good point. I plan to use a maximum and minimum value to restrict the width and height. any idea how to figure out that value?

Reply:
Puppeteer has some pre-set page dimensions that can be used as reference- if not that, then maybe using default width and height for web pages?

Reply:
1. Puppeteer default viewport size is 800 * 600 px. I think what we are discussing here is the max/min, not the default value. In my design, I set the default to be 1200 * 800, if no input is given.
2. Yeah I am thinking what's the max/min resolution for responsive web pages. Some knowledge or design convention from front-end is needed. E.g., if a user gives an input 100 * 100 px, or 10000* 10000 px, which we can't allow to happen. Clearly puppeteer doesn't have restriction on that. The image here shows the png generated using a 100 * 100 px setting.
![image](https://user-images.githubusercontent.com/32652829/86555452-3ac5ba80-bf05-11ea-8885-cabf8a75419f.png)